### PR TITLE
Adds base repository

### PR DESCRIPTION
**Pull request by DannySantos**

### CODE REVIEW

File: lib/playload/repositories/base_repository.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  def find_with(id, includes: [])
+    aggregate(*includes).where(id: id).map_to(self.class.entity).one
+  end
+
+  def find_by_with(conditions, includes: [])
+    aggregate(*includes).where(conditions).map_to(self.class.entity).one
+  end
+
+  def all_by(conditions)
+    relation_by(conditions).to_a
```
Comment (Swiftda01) [inline lib/playload/repositories/base_repository.rb]: Out of interest, what is the difference between this and `fetch_relation.where(conditions)` ?

File: spec/playload/repositories/base_repository_spec.rb [hunk 2]
```diff
@@ -1,8 +1,10 @@
 # frozen_string_literal: true
 
-RSpec.describe UserRepository, type: :repository do
+RSpec.describe BaseRepository, type: :repository do
```
Comment (Swiftda01) [inline spec/playload/repositories/base_repository_spec.rb:3]: Should we not still test that the user repository implements the base methods? Or are we going to rely on the assumption that all repositories include base repository methods?

**Review by Swiftda01 — APPROVED** [review]

File: spec/playload/repositories/base_repository_spec.rb (hunk 2 above)
Comment (DannySantos) [reply spec/playload/repositories/base_repository_spec.rb:3]: The latter. Perhaps we can write a separate test instead that just makes sure every repo has base_repo included. That would probably be a good idea actually.

File: lib/playload/repositories/base_repository.rb (hunk 1 above)
Comment (DannySantos) [reply lib/playload/repositories/base_repository.rb]: Good spot, that's a typo

**Pull request merged**